theano/gradient.py

```diff
@@ -1128,6 +1141,19 @@ def try_to_copy_if_needed(var):
             # we won't be able to post-process out the Nones if it does that
             input_grads = list(input_grads)
 
+            # Need to propagate the NullType gradients; if an input grad is
+            # not disconnected and the corresponding input is connected
+            # to at least one output whose gradient is NullType then the input
+            # grad should be NullType.
+            op_conn_pattern = _node_to_pattern(node)
```

Comment:
No need to call `_node_to_pattern(node)` again, it should have already be called here: https://github.com/carriepl/Theano/blob/scan_grad_speedup2/theano/gradient.py#L985